src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/widget/WidgetPageObject.java

```diff
@@ -54,24 +77,38 @@ public WidgetPageObject navigate(String wikiUrl) {
     return this;
   }
 
-  public WidgetPageObject createAndNavigate(String wikiUrl) {
-    return create().navigate(wikiUrl);
-  }
-
-  public boolean isLoadedOnMercury() {
-    boolean result = isTagLoadedOnMercury();
+  public boolean isLoaded() {
+    boolean result = isWidgetVisible(getWidgetIFrame(), getWidgetBody());
     PageObjectLogging.log(getTagName(), MercuryMessages.VISIBLE_MSG, result);
     return result;
   }
 
-  public boolean isLoadedOnOasis() {
-    boolean result = isTagLoadedOnOasis();
-    PageObjectLogging.log(getTagName(), MercuryMessages.VISIBLE_MSG, result);
+  public boolean areAllValidSwappedForIFrames() {
+    boolean result = getWidgetWrapperList().size() == getWidgetIFrameList().size();
+    PageObjectLogging.log(getTagName(), MercuryMessages.ALL_VALID_WIDGETS_ARE_SWAPPED_MSG, result);
     return result;
   }
 
-  public WidgetPageObject createIncorrectAndNavigate(String wikiURL) {
-    return createIncorrect().navigate(wikiURL);
+  public boolean areLoaded() {
+    boolean result = true;
+    int i = 1;
+    List<WebElement> widgetIFrameList = getWidgetIFrameList();
+
+    if (widgetIFrameList.isEmpty()) {
+      result = false;
```

Comment:
you are only setting flag to false, but still unnecessary loop operations are executed